Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp

```diff
@@ -110,8 +110,18 @@ void* OSAllocator::tryReserveAndCommit(size_t bytes, Usage usage, bool writable,
         // mprotect results in multiple references to the code region. This
         // breaks the madvise based mechanism we use to return physical memory
         // to the OS.
-        mmap(result, pageSize(), PROT_NONE, MAP_FIXED | MAP_PRIVATE | MAP_ANON, fd, 0);
-        mmap(static_cast<char*>(result) + bytes - pageSize(), pageSize(), PROT_NONE, MAP_FIXED | MAP_PRIVATE | MAP_ANON, fd, 0);
+        const auto guardPageSize = pageSize();
+        auto foreGuardResult = mmap(result, guardPageSize, PROT_NONE, MAP_FIXED | MAP_PRIVATE | MAP_ANON, fd, 0);
+        if (foreGuardResult == MAP_FAILED) {
+            munmap(result, bytes);
+            return nullptr;
+        }
+        auto aftGuardResult = mmap(static_cast<char*>(result) + bytes - guardPageSize, guardPageSize, PROT_NONE, MAP_FIXED | MAP_PRIVATE | MAP_ANON, fd, 0);
+        if (aftGuardResult == MAP_FAILED) {
+            munmap(result, bytes);
+            munmap(foreGuardResult, guardPageSize);
```

Comment:
Do we need this?

Reply:
Good point, while harmless it's also superfluous here. Removed